### PR TITLE
adding event entry to message json to distinguish between events and …

### PR DESCRIPTION
**Pull request by zalmen**
…debug messages

### CODE REVIEW

File: cmd/node/node.go [hunk 1]
```diff
@@ -503,7 +503,7 @@ func getEdIdentity() (*signing.EdSigner, error) {
 }
 
 func (app *SpacemeshApp) Start(cmd *cobra.Command, args []string) {
-	log.Info("Starting Spacemesh")
+	log.With().EventInfo("Starting Spacemesh")
```
Comment (antonlerner) [inline cmd/node/node.go]: this has no parameters, no need to use event

File: cmd/sync/sync.go [hunk 2]
```diff
@@ -158,7 +158,7 @@ func (app *SyncApp) Start(cmd *cobra.Command, args []string) {
 	}
 
 	lg.Info("%v verified layers %v", app.BaseApp.Config.P2P.NodeID, app.sync.ValidatedLayer())
-	lg.Info("sync done")
+	lg.With().EventInfo("sync done")
```
Comment (antonlerner) [inline cmd/sync/sync.go]: this also

File: miner/block_builder.go [hunk 3]
```diff
@@ -260,7 +260,7 @@ func (t *BlockBuilder) listenForTx() {
 					continue
 				}
 
-				t.Log.Info("got new tx %v", hex.EncodeToString(id[:]))
+				t.Log.With().Info("got new tx", log.TxId(hex.EncodeToString(id[:5])))
```
Comment (antonlerner) [inline miner/block_builder.go]: check if you have 5 chars at least

File: miner/block_builder.go [hunk 4]
```diff
@@ -325,11 +325,11 @@ func (t *BlockBuilder) acceptBlockData() {
 		case id := <-t.beginRoundEvent:
 			atxID, proofs, err := t.blockOracle.BlockEligible(id)
 			if err != nil {
-				t.Error("failed to check for block eligibility in layer %v: %v ", id, err)
+				t.With().Error("failed to check for block eligibility", log.LayerId(uint64(id)), log.Err(err))
```
Comment (antonlerner) [inline miner/block_builder.go:328]: IMO we can leave errors as they are

File: sync/block_listener.go [hunk 5]
```diff
@@ -93,24 +92,24 @@ func (bl *BlockListener) ListenToGossipBlocks() {
 
 func (bl *BlockListener) HandleNewBlock(blk *types.Block) bool {
 
-	bl.Log.With().Info("got new block", log.Uint64("id", uint64(blk.Id)), log.Int("txs", len(blk.TxIds)), log.Int("atxs", len(blk.AtxIds)))
+	bl.Log.With().Info("got new block", log.BlockId(uint64(blk.Id)), log.Int("txs", len(blk.TxIds)), log.Int("atxs", len(blk.AtxIds)))
```
Comment (antonlerner) [inline sync/block_listener.go:95]: we can move this to debug, or the latter at the end of the func

File: log/zap.go [hunk 6]
```diff
@@ -148,6 +178,28 @@ func (l Log) WithOptions(opts ...zap.Option) Log {
 	}
 }
 
+const event_key = "event"
+
+func (fl fieldLogger) EventInfo(msg string, fields ...Field) {
```
Comment (antonlerner) [inline log/zap.go]: you can create one function called "Event" and concate it with .Info() and so on... no need for these functions

File: miner/block_builder.go (hunk 3 above)
Comment (y0sher) [reply miner/block_builder.go]: you can do min(5, len(id))

File: log/zap.go [hunk 7]
```diff
@@ -148,6 +178,28 @@ func (l Log) WithOptions(opts ...zap.Option) Log {
 	}
 }
```
Comment (y0sher) [inline log/zap.go]:
```golang
func (l Log) Event() {
  return fl.WithFields(Bool("event", true))
}
```

File: cmd/node/node.go (hunk 1 above)
Comment (zalmen) [reply cmd/node/node.go]: what is the problem with using json here?

File: miner/block_builder.go (hunk 3 above)
Comment (zalmen) [reply miner/block_builder.go]: done

File: sync/block_listener.go (hunk 5 above)
Comment (zalmen) [reply sync/block_listener.go:95]: are you sure? Do we have another indication of how long it took to syntactically validate the block?

File: cmd/node/node.go (hunk 1 above)
Comment (zalmen) [reply cmd/node/node.go]: we need it to be a json since it is an event

File: cmd/sync/sync.go (hunk 2 above)
Comment (zalmen) [reply cmd/sync/sync.go]: we need it to be a json since it is an event

File: log/zap.go (hunk 6 above)
Comment (zalmen) [reply log/zap.go]: fixed

File: log/zap.go (hunk 7 above)
Comment (zalmen) [reply log/zap.go]: thanks

**Review by antonlerner — APPROVED** [review]

**Pull request merged**